examples/server/public/index.html

```diff
@@ -185,28 +185,38 @@ <h2 class="font-bold mb-4 ml-4">Conversations</h2>
     <dialog class="modal" :class="{'modal-open': showConfigDialog}">
       <div class="modal-box">
         <h3 class="text-lg font-bold mb-6">Settings</h3>
-        <p class="opacity-40 mb-6">Settings below are saved in browser's localStorage</p>
-        <label class="form-control mb-2">
-          <div class="label">System Message</div>
-          <textarea class="textarea textarea-bordered h-24" :placeholder="'Default: ' + configDefault.systemMessage" v-model="config.systemMessage"></textarea>
-        </label>
-        <template v-for="key in Object.keys(config)">
-          <label v-if="key != 'custom' && key != 'systemMessage'"
-            class="input input-bordered flex items-center gap-2 mb-2">
-            <b>{{ key }}</b>
-            <input type="text" class="grow" :placeholder="'Default: ' + (configDefault[key] || 'none')" v-model="config[key]" />
+        <div class="h-[calc(90vh-12rem)] overflow-y-auto">
+          <p class="opacity-40 mb-6">Settings below are saved in browser's localStorage</p>
+          <label class="form-control mb-2">
+            <div class="label">System Message</div>
+            <textarea class="textarea textarea-bordered h-24" :placeholder="'Default: ' + configDefault.systemMessage" v-model="config.systemMessage"></textarea>
           </label>
-        </template>
-        <label class="form-control mb-2">
-          <div class="label inline">Custom JSON config (For more info, refer to <a class="underline" href="https://github.com/ggerganov/llama.cpp/blob/master/examples/server/README.md" target="_blank" rel="noopener noreferrer">server documentation</a>)</div>
-          <textarea class="textarea textarea-bordered h-24" placeholder="Example: { &quot;mirostat&quot;: 1, &quot;min_p&quot;: 0.1 }" v-model="config.custom"></textarea>
-        </label>
+          <template v-for="key in ['temperature', 'top_k', 'top_p', 'min_p', 'max_tokens']">
+            <label class="input input-bordered flex items-center gap-2 mb-2">
+              <b>{{ key }}</b>
+              <input type="text" class="grow" :placeholder="'Default: ' + (configDefault[key] || 'none')" v-model="config[key]" />
```

Comment:
@MaggotHATE I ran into the issue with the input types as well (using Chrome) and I think this might be a cause of it.

Reply:
Should be fixed in https://github.com/ggerganov/llama.cpp/pull/10175/commits/f2268fad8189ec99eb83df93c81d8676a96a875e

For an `<input>`, they value will always be a string (no matter if `type="number"`), so the check need to be "is string convertible to number" instead of "isNumeric"

Reply:
With the `<input>` set to `type="number"` the type does seem to be number instead of string when I print out the type:
```js
 console.log("type of key", key, typeof this.config[key]);
```
And the output in the browser console is:
```console
type of key top_p number
```
But your fixes work so :+1:

Reply:
Hmm interesting find. Normally, `HTMLInputElement.value` is [always a string](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/value). Maybe vuejs have something on top to handle the case where `type="number"`. But I think I'll look into that in the future.